### ISSUE TITLE
BUG: `empty_like` should fill object references with `None`

### ISSUE THREAD

**Issue by seberg** [numpy/numpy#21738]
Pandas ran into this in [here](https://github.com/pandas-dev/pandas/pull/47097/files#r895218281).  Normally we fill empty object arrays with `None`.
NumPy supports it being filled with `NULL` just fine (and sometimes that should happen before any other initialization took place).

But, generally, NumPy *should* always fill with None and it would be nice to make "enforce" this (i.e. remove support for `NULL` in a few places, because only bad user-provided arrays would use it.

**TL;DR:** We should ensure that `np.empty_like` fills object arrays the same way that `np.empty` does.  This can be tested using `arr.tobytes()`.

**Comment by seberg** [comment]
Ah, this was previously reported in gh-7854 and there was even a PR.  I still think we should just align `empty_like` and aim for `NULL` being considered a wart that may eventually go away (for fully initialized arrays).
